### ISSUE TITLE
SPARQLQueryRenderer Does Not Support Update/Insert (ParsedUpdate)

### ISSUE THREAD

**Issue by daltontc** [eclipse/rdf4j#2766]
The SPARQLQueryRenderer should be extended out to handle parsing INSERT queries. Right now it only supports rendering strings for ParsedQuery objects and not ParsedUpdate object. Is there another way to get the query string from a ParsedUpdate?

**Comment by jeenbroekstra** [comment]
You can get the original string that produced the ParsedUpdate back using `ParsedUpdate#getSourceString()`.

**Comment by daltontc** [comment]
Right, but if I want to provide a baseURI along with the query and get the fully formed query with the baseURI substituted in, will that method do that? I thought it only provided the original string.

**Comment by jeenbroekstra** [comment]
> Right, but if I want to provide a baseURI along with the query and get the fully formed query with the baseURI substituted in, will that method do that? I thought it only provided the original string.

You're correct. I'm not sure there is an easy way to convert that back to a string represention at present, sorry. Can I ask what it is you're trying to do / why you need this?

**Comment by daltontc** [comment]
Our system usually interfaces with query engines that use the standard RDF4j RepositoryConnection, but I have this separate one that we are trying to plug into that just takes a string for the whole query. We are trying to wrap connecting to it using the same RepositoryConnection API. I have been successful in getting every method working other than the `prepareUpdate(QueryLanguage ql, String update, String baseURI)`. I just wanted to be able to use the same API but have it transform the query into a query string with the properly substituted baseURI for when we use that in other services.

**Comment by jeenbroekstra** [comment]
You could just do what the SPARQLConnection does, which is just to prepend the source update string with a base clause:

```java
String update = "BASE <" + baseURI + "> \n" + update;
```

Not the most elegant solution but it will work I think.
